cloud/amazon/ec2_vpc_net.py

```diff
@@ -231,7 +234,9 @@ def main():
 
         if vpc_obj is None:
             try:
-                vpc_obj = connection.create_vpc(cidr_block, instance_tenancy=tenancy)
+                if not module.check_mode:
+                    vpc_obj = connection.create_vpc(cidr_block, instance_tenancy=tenancy)
+
                 changed = True
```

Comment:
since you're not implementing diff mode, probably best to move the ```changed``` set above and bail here with an else- otherwise you'll need to check for null vpc_obj all over the place in the downstream checks.